Signal/src/ViewControllers/HomeView/Chat List/ChatListViewController.swift

```diff
@@ -496,6 +496,32 @@ public extension ChatListViewController {
         navigationController.setViewControllers(viewControllers, animated: false)
         presentFormSheet(navigationController, animated: true, completion: completion)
     }
+
+    /// Verifies that the currently selected cell matches the provided thread's uniqueId.
+    /// If it does or if the user's in multi-select: Do nothing.
+    /// If it doesn't: Select the first cell matching the provided thread, if one exists. Otherwise, deselect the current row.
+    @objc
+    func ensureSelectedThread(_ targetThread: TSThread, animated: Bool) {
+        // Ignore any updates if we're in multiselect mode. I don't think this can happen,
+        // but if it does let's avoid stepping over the user's manual selection.
+        let currentSelection = tableView.indexPathsForSelectedRows ?? []
+        guard viewState.multiSelectState.isActive == false, currentSelection.count < 2 else {
+            return
+        }
+
+        let currentlySelectedThread = currentSelection.first.flatMap {
```

Comment:
nit: since if this ends up `nil` we'll basically just return anyway, would prefer we do above as:

```swift
guard
    !viewState.multiSelectState.isActive,
    currentSelection.count < 2,
    let currentlySelectedThread = currentSelection.first
else {
    return
}

let ... = self.tableDataSource.thread(forIndexPath...)
```

or similar.

Reply:
Can then reuse `currentlySelectedThread` below, and not need an `else if`

Reply:
> nit: since if this ends up `nil` we'll basically just return anyway, would prefer we do above as:

We may need to do some work if this ends up being nil. If there's no currently selected thread but the target thread is in the render state, then:
- `currentlySelectedThread?.uniqueId != targetThread.uniqueId` returns true
- `if let targetPath = tableDataSource.renderState.indexPath(...` returns a valid path
- The targetPath gets selected

Reply:
Ah, yeah I mixed up my `==` and `!=`. We can leave this as-is I think.